src/platform/plugins/private/presentation_panel/public/panel_component/panel_header/presentation_panel_hover_actions.tsx

```diff
@@ -130,106 +131,26 @@ export const PresentationPanelHoverActions = ({
-      (dragHandleRef.current?.offsetWidth ?? 0) +
-      parseInt(euiTheme.size.base, 10) * 2;
-    const hoverActionsHeight = rightHoverActionsRef.current?.offsetHeight ?? 0;
-
-    // Left align hover actions when they would get cut off by the right edge of the window
-    if (anchorLeft - (hoverActionsWidth - anchorWidth) <= parseInt(euiTheme.size.base, 10)) {
-      dragHandleRef.current?.style.removeProperty('right');
-      dragHandleRef.current?.style.setProperty('left', '0');
-    } else {
-      hoverActionsRef.current.style.removeProperty('left');
-      hoverActionsRef.current.style.setProperty('right', '0');
-    }
-
-    if (anchorRef.current && rightHoverActionsRef.current) {
-      const shouldCombine = anchorWidth < hoverActionsWidth;
-      const willGetCutOff = anchorTop < hoverActionsHeight;
-
-      if (shouldCombine !== combineHoverActions) {
-        setCombineHoverActions(shouldCombine);
-      }
-
-      if (willGetCutOff) {
-        hoverActionsRef.current.style.setProperty('position', 'absolute');
-        hoverActionsRef.current.style.setProperty('top', `-${euiTheme.size.s}`);
-      } else if (shouldCombine) {
-        hoverActionsRef.current.style.setProperty('top', `-${euiTheme.size.l}`);
-      } else {
-        hoverActionsRef.current.style.removeProperty('position');
-        hoverActionsRef.current.style.removeProperty('top');
-      }
-
-      if (shouldCombine || willGetCutOff) {
-        setBorderStyles(ALL_ROUNDED_CORNERS);
-      } else {
-        setBorderStyles(TOP_ROUNDED_CORNERS);
-      }
-    }
-  };
-
-  const [
-    defaultTitle,
-    title,
-    description,
-    hidePanelTitle,
-    hasLockedHoverActions,
-    parentHideTitle,
-    parentViewMode,
-  ] = useBatchedOptionalPublishingSubjects(
-    api?.defaultTitle$,
-    api?.title$,
-    api?.description$,
-    api?.hideTitle$,
-    api?.hasLockedHoverActions$,
-    api?.parentApi?.hideTitle$,
-    /**
-     * View mode changes often have the biggest influence over which actions will be compatible,
-     * so we build and update all actions when the view mode changes. This is temporary, as these
-     * actions should eventually all be Frequent Compatibility Change Actions which can track their
-     * own dependencies.
-     */
-    getViewModeSubject(api ?? undefined)
-  );
+  const [defaultTitle, title, description, hidePanelTitle, hasLockedHoverActions, parentHideTitle] =
+    useBatchedOptionalPublishingSubjects(
+      api?.defaultTitle$,
+      api?.title$,
+      api?.description$,
+      api?.hideTitle$,
+      api?.hasLockedHoverActions$,
+      api?.parentApi?.hideTitle$
+    );
 
   const hideTitle = hidePanelTitle || parentHideTitle;
   const showDescription = description && (!title || hideTitle);
 
   const quickActionIds = useMemo(
-    () => QUICK_ACTION_IDS[parentViewMode === 'edit' ? 'edit' : 'view'],
-    [parentViewMode]
+    () => QUICK_ACTION_IDS[viewMode === 'edit' ? 'edit' : 'view'],
```

Comment:
We were sending in `viewMode` as a prop **and** getting the value via `getViewModeSubject` - this resulted in a double render.